ports/stm32/machine_i2c.c

```diff
@@ -44,28 +44,29 @@ typedef struct _machine_hard_i2c_obj_t {
     i2c_t *i2c;
     mp_hal_pin_obj_t scl;
     mp_hal_pin_obj_t sda;
+    uint32_t timeout;
 } machine_hard_i2c_obj_t;
 
 STATIC const machine_hard_i2c_obj_t machine_hard_i2c_obj[] = {
     #if defined(MICROPY_HW_I2C1_SCL)
     {{&machine_hard_i2c_type}, I2C1, MICROPY_HW_I2C1_SCL, MICROPY_HW_I2C1_SDA},
```

Comment:
This would become: `{{&machine_hard_i2c_type}, I2C1, MICROPY_HW_I2C1_SCL, MICROPY_HW_I2C1_SDA, &timeout_array[0]}`